app/components/network_indicator/network_indicator.js

```diff
@@ -0,0 +1,390 @@
+    Platform,
+    StyleSheet,
+    TouchableOpacity,
+    View,
+} from 'react-native';
+import DeviceInfo from 'react-native-device-info';
+import IonIcon from 'react-native-vector-icons/Ionicons';
+
+import FormattedText from 'app/components/formatted_text';
+import {ViewTypes} from 'app/constants';
+import mattermostBucket from 'app/mattermost_bucket';
+import networkConnectionListener, {checkConnection} from 'app/utils/network';
+import {t} from 'app/utils/i18n';
+import LocalConfig from 'assets/config';
+
+import {RequestStatus} from 'mattermost-redux/constants';
+
+const HEIGHT = 38;
+const OFFLINE = 'offline';
+const CONNECTING = 'connecting';
+const CONNECTED = 'connected';
+const {
+    ANDROID_TOP_LANDSCAPE,
+    ANDROID_TOP_PORTRAIT,
+    IOS_TOP_LANDSCAPE,
+    IOS_TOP_PORTRAIT,
+    IOSX_TOP_PORTRAIT,
+    STATUS_BAR_HEIGHT,
+} = ViewTypes;
+
+export default class NetworkIndicator extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            closeWebSocket: PropTypes.func.isRequired,
+            connection: PropTypes.func.isRequired,
+            initWebSocket: PropTypes.func.isRequired,
+            logout: PropTypes.func.isRequired,
+            startPeriodicStatusUpdates: PropTypes.func.isRequired,
+            stopPeriodicStatusUpdates: PropTypes.func.isRequired,
+        }).isRequired,
+        isConnecting: PropTypes.bool,
+        isLandscape: PropTypes.bool,
+        isOnline: PropTypes.bool,
+        webSocketStatus: PropTypes.string,
+    };
+
+    static defaultProps = {
+        isOnline: true,
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+        const {isLandscape, networkStatus} = prevState;
+
+        if (nextProps.isLandscape !== isLandscape && networkStatus) {
+            const navBar = this.getNavBarHeight(nextProps.isLandscape);
+            const top = new Animated.Value(navBar - HEIGHT);
+
+            return {
+                navBar,
+                top,
+            };
+        }
+
+        return null;
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.isX = DeviceInfo.getModel().includes('iPhone X');
+        const navBar = this.getNavBarHeight(props.isLandscape);
+
+        this.state = {
+            isLandscape: props.isLandscape,
+            networkStatus: null,
+            navBar,
+            top: new Animated.Value(navBar - HEIGHT),
+        };
+
+        this.backgroundColor = new Animated.Value(0);
+
+        this.networkListener = networkConnectionListener(this.handleConnectionChange);
+    }
+
+    componentDidMount() {
+        this.mounted = true;
+    }
+
+    componentDidUpdate(prevProps) {
+        const {webSocketStatus} = this.props;
+
+        if (this.props.isOnline) {
+            if (this.state.networkStatus && webSocketStatus === RequestStatus.STARTED && this.props.webSocketStatus === RequestStatus.SUCCESS) {
```

Comment:
I think `webSocketStatus` and `this.props.webSocketStatus` here is the same one, isn't it?

Reply:
Yep, you're right. Fixed.